Content.Server/Medical/SuitSensors/SuitSensorSystem.cs

```diff
@@ -287,6 +300,35 @@ private string GetModeName(SuitSensorMode mode)
         return Loc.GetString(name);
     }
 
+    public void TrySetSensor(EntityUid uid, SuitSensorMode mode, EntityUid userUid,
+        SuitSensorComponent? component = null)
+    {
+        if (!Resolve(uid, ref component))
+            return;
+
+        if (component.User == null || userUid == component.User)
+            SetSensor(uid, mode, userUid, component);
+        else
+        {
+            var doAfterEvent = new SuitSensorChangeDoAfterEvent(mode);
+            var doAfterArgs = new DoAfterArgs(EntityManager, userUid, component.SensorsTime, doAfterEvent, uid)
+            {
+                BreakOnMove = true,
+                BreakOnDamage = true
+            };
+
+            _doAfterSystem.TryStartDoAfter(doAfterArgs);
+        }
+    }
+
+    private void OnSuitSensorDoAfter(EntityUid uid, SuitSensorComponent component, SuitSensorChangeDoAfterEvent args)
```

Comment:
Similar to before, use `Entity<SuitSensorComponent>`

Reply:
Because of SubscribeLocalEvent it can't work properly with Entity<>

Dunno why, but that always with most events

Reply:
You need to do `Entity<blabla>, ref Event args`, the ref in front of the second argument is what fixes it

Reply:
> You need to add `ref` before `SuitSensorChangeDoafterEvent`

Thanks, it helped